onnx/defs/tensor/defs.cc

```diff
@@ -536,96 +537,156 @@ ONNX_OPERATOR_SET_SCHEMA(
             return;
           }
 
-          auto getInitializerData =
+          // don't know data_type- can't proceed
+          if (!startsInitializer->has_data_type())
+            return;
+
+          auto get_initializer_data =
               [](const TensorProto* initializer) -> std::vector<int64_t> {
             std::vector<int64_t> vec;
-            if (initializer->has_raw_data()) {
+            if (initializer->has_raw_data() &&
+                initializer->data_type() == TensorProto::INT64) {
               const std::string& bytes = initializer->raw_data();
               vec.insert(
                   vec.end(),
                   reinterpret_cast<const int64_t*>(bytes.c_str()),
                   reinterpret_cast<const int64_t*>(
                       bytes.c_str() + bytes.size()));
-            } else {
+            } else if (
+                initializer->has_raw_data() &&
```

Comment:
It may have byte-order issue.

Reply:
you mean endianness of the machine we are trying to parse the raw data ? That's a good point.  Thanks. I will fix it.

Reply:
You may use [this](https://github.com/Microsoft/onnxruntime/blob/master/onnxruntime/core/framework/tensorprotoutils.cc#L49) as a reference.

Reply:
Oh I didn't see your comment - I implemented it already and tested it. You may take a look.